modules/core/include/opencv2/core/mat.hpp

```diff
@@ -358,13 +361,15 @@ class CV_EXPORTS _InputOutputArray : public _OutputArray
     template<typename _Tp, int m, int n> _InputOutputArray(const Matx<_Tp, m, n>& matx);
     _InputOutputArray(const UMat& m);
     _InputOutputArray(const std::vector<UMat>& vec);
+
+    explicit _InputOutputArray(const _OutputArray& o);
 };
 
-typedef const _InputArray& InputArray;
+typedef _InputArray InputArray;
```

Comment:
@alalek, you are right, I will revert the change. I did this change mainly out of curiosity, to see how much code depend on such a _InputArray&lt;=&gt;InputArray relationship. Apparently, there are very few changes needed. But I agree that better source-level compatibility with 2.4 is more important than questionable gain in speed.